### ISSUE TITLE
client.files.getReadStream gets stuck when data is not immediately read

### ISSUE THREAD

**Issue by bskiff** [box/box-node-sdk#763]
- [x] I have checked that the [SDK documentation][sdk-docs] doesn't solve my issue.
- [x] I have checked that the [API documentation][api-docs] doesn't solve my issue.
- [x] I have searched the [Box Developer Forums][dev-forums] and my issue isn't already reported (or if it has been reported, I have attached a link to it, for reference).
- [x] I have searched [Issues in this repo][github-repo] and my issue isn't already reported.

### Description of the Issue
When data is not read synchronously from the read stream returned from `client.files.getReadStream`, the data is never provided and the stream is not closed.

The issue was already reported in #704, but was not resolved. It is still an issue, and hopefully our examples/workarounds can provide some additional insight. 

### Steps to Reproduce
```javascript
const sdkInstance = BoxSDK.getPreconfiguredInstance(config);
const client = sdkInstance.getAppAuthClient('enterprise');

const fileStream = await client.files.getReadStream(fileId);

setTimeout(() => {
  // data never loads!
  fileStream.pipe(process.stdout);
}, 1000);
```

We were able to workaround this issue by adding in a `Passthrough` that doesn't do anything.
```javascript
const sdkInstance = BoxSDK.getPreconfiguredInstance(config);
const client = sdkInstance.getAppAuthClient('enterprise');

const fileStream = await client.files.getReadStream(fileId);

const passThrough = new PassThrough();
fileStream.pipe(passThrough);

setTimeout(() => {
  // data loads!
  passThrough.pipe(process.stdout);
}, 1000);
```

### Expected Behavior
Data is never made available on the stream, if the first read from the stream occurs asynchronously.

### Versions Used
Node SDK: 2.5.0, 2.6.0

[sdk-docs]: ./doc
[api-docs]: https://developer.box.com/docs
[dev-forums]: https://community.box.com/t5/Platform-and-Development-Forum/bd-p/DeveloperForum
[github-repo]: https://github.com/box/box-node-sdk/search?type=Issues

**Comment by arjankowski** [comment]
Hi @bskiff,

Thanks for your insights! 
I've created a jira ticket for my team (`jira-2645`), so we can look into this soon.

@arjankowski

**Comment by congminh1254** [comment]
Hi @bskiff,

We have just fixed this issue using `PassThrough`, and it will be available in the next Box Node SDK release.

Thank you for your solution, and if you have any other problem, feel free to let us know.

Regards,
Minh

**Comment by bskiff** [comment]
Thanks @congminh1254!

**Comment by congminh1254** [comment]
Hi @bskiff ,

We did a release contains your bug fix, you can use Node SDK 2.8.1 to verify if it works for you.

Regards,
Minh

**Comment by bskiff** [comment]
We pulled in the new version, and it's working! Thanks again @congminh1254!
